bridges/BookMyShowBridge.php

```diff
@@ -0,0 +1,1388 @@
+			// Sample Output = ['Adventure', 'Animation', 'Comedy']
+			'categories' => array_filter(
+				explode('|', ucwords(strtolower($eventGroup['EventGrpGenre']), '|'))
+			),
+			'uid' => $eventGroup['EventGroup']
+		);
+	}
+
+	private function generateEventData($event, $category){
+		if($category == self::MOVIES) {
+			return $this->generateMoviesData($event);
+		}
+
+		return $this->generateGenericEventData($event, $category);
+	}
+
+	/**
+	 * Takes an event data as array and returns data for RSS Post
+	 */
+	private function generateGenericEventData($event, $category){
+		return array(
+			'uri' => $event['FShareURL'],
+			'title' => $event['EventTitle'],
+			'timestamp' => $event['Event_dtmCreated'],
+			'author' => 'BookMyShow',
+			'content' => $this->generateEventHtml($event, $category),
+			'enclosures' => array(
+				$event['BannerURL'],
+			),
+			'categories' => array_merge(
+				array(self::CATEGORIES[$category]),
+				$event['GenreArray']
+			),
+			'uid' => $event['EventGroupCode']
+		);
+	}
+
+	public function collectData(){
+		$city = $this->getInput('city');
+		$category = $this->getInput('category');
+
+		$url = $this->makeUrl($category);
+		$headers = $this->makeHeaders($city);
+
+		$data = json_decode(getContents($url, $headers), true);
+
+		if ($category == self::MOVIES) {
+			$data = $data['moviesData']['BookMyShow']['arrEvents'];
+		} else {
+			$data = $data['data']['BookMyShow']['arrEvent'];
+		}
+
+		foreach ($data as $event) {
+			$item = $this->generateEventData($event, $category);
+			if ($this->matchesFilters($category)) {
+				$this->items[] = $item;
+			}
+		}
+	}
+
+	/**
+	 * Currently only checks if the language filter matches
+	 */
+	private function matchesFilters($category){
+		if ($this->getInput('language') !== 'all') {
+			$language = $this->getInput('language');
+			return in_array($language, $this->languages);
+		}
+		return true;
+	}
+
+	public function getName(){
+		$city = $this->getInput('city');
+		$category = $this->getInput('category');
+		if(!is_null($city) and !is_null($category)) {
+			$categoryName = self::CATEGORIES[$category];
+			$cityNames = array_flip(self::CITIES);
+			$cityName = $cityNames[$city];
+			return "BookMyShow: $categoryName in $cityName";
+		}
+
+		return parent::getName();
+	}
+
+	/**
+	 * Returns
+	 * @param  string $city City Code
+	 * @return array list of headers
+	 */
+	private function makeHeaders($city){
+		$uniqid = uniqid();
+		$rgn = urlencode("|Code=$city|");
+		return array(
+			"Cookie: bmsId=$uniqid; Rgn=$rgn;"
+		);
+	}
+
+	/**
+	 * Generates various URLs as per https://tools.ietf.org/html/rfc5870
+	 * and other standards
+	 */
+	private function generateDirectionsHtml($lat, $long, $address = ''){
+		$address = urlencode($address);
+
+		$links = array(
+			'Apple Maps' => 'http://maps.apple.com/maps?q=%s,%s"',
+			'Google Maps' => 'http://maps.google.com/maps?ll=%s,%s',
+			// 'Google Maps (Android)' => 'geo:%s,%s?q=%s',
```

Comment:
I spent wayy too much time generating all these format strings to remove them from code entirely.

5 entries was taking up too much space, so I reduced it to 3 and commented the optional Google Maps ones.

Reply:
Instead of a list you could use a comma-separated list to make it smaller (starting at line 1380 below):

```php
		$html = '<h4>Directions</h4><p>';
		$locations = array();

		foreach ($links as $app => $str) {
			$url = sprintf($str, $lat, $long, $address);
			$locations[] = "<a href='$url' title='$app'>$app</a>";
		}

		$html .= implode(', ', $locations) . '</p>';
```